staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -401,6 +401,7 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 func (r *Reflector) ListAndWatchWithContext(ctx context.Context) error {
 	logger := klog.FromContext(ctx)
 	logger.V(3).Info("Listing and watching", "type", r.typeDescription, "reflector", r.name)
+	defer utilruntime.HandleCrashWithContext(ctx)
```

Comment:
we didn't log panics in this method before, I'm not really sure why this level is where we'd choose to start logging them now

Reply:
I agree.

@ballista01: with "leave this to the caller" in https://github.com/kubernetes/kubernetes/pull/131000#discussion_r2066197963 I meant the caller of client-go, not a higher-level function in client-go. Please don't resolve comments without giving your reviewer a chance to check whether their comment really is addressed.